### PR TITLE
Mappings for 21w41a world generation

### PR DESCRIPTION
**Pull request by mschae23**
Started adding mappings for world generation changes in 21w41a, including the new "material rules", which replaced surface builders.

### CODE REVIEW

**Comment by mschae23** [comment]
This is now ready for review.

File: mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+			ARG 1 biomeKey
+			ARG 2 biome
+			ARG 3 runDepth
+			ARG 4 stoneDepthAbove
+			ARG 5 stoneDepthBelow
+			ARG 6 waterHeight
+			ARG 7 x
+			ARG 8 y
+			ARG 9 z
+		CLASS class_6695 LessThanOrEqualPredicate
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/surfacebuilder/MaterialRules.mapping]:
```suggestion
		CLASS class_6695 NegativePredicate
```

File: mappings/net/minecraft/world/gen/NoiseColumnSampler.mapping [hunk 2]
```diff
@@ -6,30 +6,94 @@ CLASS net/minecraft/class_5818 net/minecraft/world/gen/NoiseColumnSampler
...
+	FIELD field_34662 oreRandomDeriver Lnet/minecraft/class_6574;
+	FIELD field_34663 noodleNoiseFactory Lnet/minecraft/class_6568$class_6571;
+	FIELD field_34664 noodleThicknessNoiseFactory Lnet/minecraft/class_6568$class_6571;
+	FIELD field_34665 noodleRidgeFirstNoiseFactory Lnet/minecraft/class_6568$class_6571;
+	FIELD field_34666 noodleRidgeSecondNoiseFactory Lnet/minecraft/class_6568$class_6571;
 	FIELD field_34667 hasNoiseCaves Z
-	FIELD field_34680 noise Lnet/minecraft/class_5822;
+	FIELD field_34680 terrainNoise Lnet/minecraft/class_5822;
+	FIELD field_34681 jaggedNoise Lnet/minecraft/class_5216;
+	FIELD field_34682 minimumY I
```
Comment (enbrain) [inline mappings/net/minecraft/world/gen/NoiseColumnSampler.mapping]:
```suggestion
	FIELD field_34682 minY I
```

**Review by enbrain — APPROVED** [review]

**Comment by modmuss50** [comment]
Looks good, too much for me to a full review, can always improve this if needed 👍

**Pull request merged**